.github/workflows/ci.yml

```diff
@@ -0,0 +1,155 @@
+        with:
+          base: master
+          # `-helm` suffix aims that we can easily filter helm charts
+          filters: |
+            hydra-helm:
+              - 'helm/charts/hydra/**'
+            hydra-maester-helm:
+              - 'helm/charts/hydra-maester/**'
+            keto-helm:
+              - 'helm/charts/keto/**'
+            kratos-helm:
+              - 'helm/charts/kratos/**'
+            kratos-selfservice-ui-node-helm:
+              - 'helm/charts/kratos-selfservice-ui-node/**'
+            oathkeeper-helm:
+              - 'helm/charts/oathkeeper/**'
+            oathkeeper-maester-helm:
+              - 'helm/charts/oathkeeper-maester/**'
+      # This step will take output from paths-filter and then process it in order to get what helm charts have been updated
+      # It allows us to do matrix in validating & testing process
+      - name: Generate helm chart matrix to be validated
+        id: changed_charts
+        env:
+          FILTER_OUTPUT: ${{ toJson(steps.filter.outputs) }}
+        run: |
+           updated_charts=$(echo "$FILTER_OUTPUT" | jq -r 'to_entries | map(select((.key | endswith("-helm")) and .value == "true")) | map(.key)')
+           updated_charts=$(echo "$updated_charts" | sed "s/-helm//g")
+
+           if [[ "$updated_charts" == "[]" ]]; then 
+            echo ::set-output name=no_changes::"true"
+           fi
+           echo ::set-output name=matrix::{\"chart\":$(echo $updated_charts)}\"
+           
+           echo "Charts array to run CI on: $updated_charts"
+  
+  check:
+    name: Check Helm Chart '${{ matrix.chart }}'
+    if: ${{ needs.detect-repo-changes.outputs.no-changes != 'true' }}
+    needs: [dependencies, detect-repo-changes]
+    runs-on: ubuntu-20.04
+    env:
+      HELM_CHART: ${{ matrix.chart }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Installing kubeval
+        run: helm plugin install https://github.com/instrumenta/helm-kubeval
+      - name: Lint helm chart 
+        run: make helm-lint
+      - name: Validate helm chart 
+        run: make helm-validate
+    strategy:
+      matrix: ${{ fromJson(needs.detect-repo-changes.outputs.helms-changed) }}
+
+ 
+  # test-upgrade:
+  #   name: Upgrade Helm Chart '${{ matrix.chart }}'
+  #   if: ${{ needs.detect-repo-changes.outputs.no-changes != 'true' }}
+  #   needs: [check, detect-repo-changes, dependencies]
+  #   runs-on: ubuntu-20.04
+  #   env:
+  #     HELM_CHART: ${{ matrix.chart }}
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v2
+  #     - name: Install k3d 
+  #       run: |
+  #         export PATH=".bin:$PATH"
+  #         curl -s https://raw.githubusercontent.com/k3d-io/k3d/v5.4.4/install.sh | USE_SUDO="false" K3D_INSTALL_DIR=".bin" bash
+  #     - name: Test upgrade for helm chart 
+  #       run: make helm-upgrade
+  #   strategy:
+  #     matrix: ${{ fromJson(needs.detect-repo-changes.outputs.helms-changed) }}
```

Comment:
I have put in comment this part, as we cannot make it work right now as there are breaking changes on scripts & helm values for CI. Here what it would do if we run it right now:

```
- Pipeline runs, generate test-upgrade matrix of jobs
- Each test-upgrade jobs tries to fetch script /hacks/values/${HELM_CHART} from the previous version of the chart. <-- At this point it fails because /hacks doesn't exist 
```

This is due to a move of helm CI values and scripts into folder `/hacks`. We will be able to uncomment this job after the next release of charts